### ISSUE TITLE
Invalid locale en-CA on MacOS Sierra (conversion needed to en_CA)

### ISSUE THREAD

**Issue by hcorion** [godotengine/godot#6942]
**Operating system or device - Godot version:**
OS: MacOS Sierra 10.12.1
Godot version: latest git
**Issue description** (what happened, and what was expected):
When running godot engine I noticed I got the error: 
```
ERROR: set_locale: Invalid Locale: en-CA
   At: core/translation.cpp:910.
```
en_CA is a valid locale though. MacOS Sierra (possibly older version I don't know) seems to use a different locale naming scheme that screws the detection up. I also noticed that in issue #6301's log it mentioned that `set_locale: Invalid Locale: en-US` further proving my point.
**Steps to reproduce:**
Build godot engine on MacOS sierra, run it, observe log for Invalid Locale errors.

**Comment by akien-mga** [comment]
Apple, the Art of Compatibility Breakage™.

Guess we need to add a conversion system to `s/_/-/g` in locale identifiers on MacOS Sierra...

**Comment by hcorion** [comment]
I'm sure Apple had a very good reason the change the locale from en_US to en-US 😛.

**Comment by damon-myers** [comment]
@akien-mga: I think #6910 does somewhat workaround this issue, do you think it's still worth implementing a conversion for MacOS Sierra?

**Comment by bojidar-bg** [comment]
@damon-myers I'd say, yes :smile:

**Comment by akien-mga** [comment]
Yes, if we have an `es_AR` translation, `es-AR` should use it, not default to `es`.

**Comment by akien-mga** [comment]
BTW if you want to implement this conversion, don't hesitate to do it all in one commit ;) Your three commits in #6910 were a bit too atomic for my tastes, I squashed them before merging.

**Comment by damon-myers** [comment]
Okay, okay. Haha. I honestly don't have as much experience with localization so I didn't understand the significance.

Sure thing, I'll take care of this and do that this time! Thanks!

**Comment by akien-mga** [comment]
Regarding the fix, I think the most straightforward way would be to just do:
```
p_locale = p_locale.replace("-", "_");
```
before using the `p_locale`.

**Comment by Keyaku** [comment]
I can confirm I had this issue with at least 10.11 (El Capitan) too, mine being pt-PT.

I just ignored it thinking it was an issue with my locale, completely ignoring that pt_PT was available and that the problem was with the hyphen/underscore.

**Comment by damon-myers** [comment]
@akien-mga Good idea, thanks for that. Don't think I could reassign to p_locale, so just created a local variable for it. Seems to have worked. :)

**Comment by akien-mga** [comment]
@Keyaku Ah interesting to now that it's not a recent Revolution™ then, weird that it never was reported before as it should have affected every single Mac user (at least with 10.11+).

**Comment by akien-mga** [comment]
Fixed by #6943.
